### ISSUE TITLE
Memory leaking on SegmentationInterpretation.from_learner

### ISSUE THREAD

**Issue by tiagojc** [fastai/fastai#3266]
I was trying to use SegmentationInterpretation.from_learner to understand my segmentation model but looks there is a memory leaking on this function. I did run it in a machine with 64GB + 10 Swap and even this way process crashed after consume the entire memory. My training set has 31652 images and my validation set has 4264 images, all of them with dimensions 800x600

**Comment by tcysin** [comment]
I'm facing the same problem, and it seems like this is related to #3141.

**Comment by warner-benjamin** [comment]
This issue should be resolved by #3558.
